src/prefect/task_worker.py

```diff
@@ -448,7 +456,10 @@ def yell(message: str):
             serve(say, yell)
         ```
     """
-    task_worker = TaskWorker(*tasks, limit=limit)
+
+    task_worker = run_coro_as_sync(_make_task_worker(*tasks, limit=limit))
```

Comment:
this is a bit of a hack to avoid moving the creation of the limiter and anyio task groups from `TaskWorker.__init__` to `TaskWorker.__aenter__`, which appears to cause problems

open to other suggestions here

Reply:
I'd have to investigate deeper but this seems like a smell to me that we should investigate - maybe add a TODO comment?

Reply:
fixed this in [997f782](https://github.com/PrefectHQ/prefect/pull/14855/commits/997f7824cfd1d57c298210b257550a5b591aa45b)